.github/workflows/main.yaml

```diff
@@ -15,54 +15,46 @@ jobs:
     steps:
       - uses: actions/checkout@v4
 
-      - name: Set up Python 3.10
-        uses: actions/setup-python@v5
-        with:
-          python-version: "3.10"
-
-      - name: Get Ruff Version
-        id: get_ruff_version
-        uses: mikefarah/yq@master
-        with:
-          cmd: yq -oy '.tool.poetry.dev-dependencies.ruff' pyproject.toml
-
-      - name: Install uv, venv, Ruff
+      - name: Set up Environment
         id: setup
-        env:
-          RUFF_VERSION: ${{ steps.get_ruff_version.outputs.result }}
         run: |
           curl -LsSf https://astral.sh/uv/install.sh | sh
           uv venv
-          uv pip install ruff
           echo "VIRTUAL_ENV=$HOME/.venv" >> $GITHUB_ENV
         continue-on-error: false
 
+      - name: Install Package
```

Comment:
Since the ruff version parsing was not working, I'm just installing the full thing before linting. It's quicker with `uv` at least 🤷

Reply:
this is due to changing the format from `ruff = "0.2.1"` to the ` ruff = [ruff...]` i take it? YQ should be able to parse that, can test in a terminal.

Reply:
<img width="774" alt="image" src="https://github.com/mozilla-ai/lm-buddy/assets/6970170/a54d933f-f30d-45a0-9c80-f373193a5158">
works locally with this changeset. what was wrong?

Reply:
I updated the `yq` command to work with the new pyproject format. 

The `steps.get_ruff_version.outputs.result` value is `null` between steps, and I don't know why.

The `RUFF_VERSION` env var was not being used before my changes, which is why we missed that.

Reply:
Hmm, the `null` appears to be the output from `yq` for a no-match.

I wonder if we are missing the match for some reason when running in GH actions.